test/unit/math/fwd/mat/vectorize/expect_val_deriv_eq.hpp

```diff
@@ -3,8 +3,11 @@
 
 #include <stan/math/fwd/mat.hpp>
 #include <gtest/gtest.h>
+#include <stan/math/prim/scal/fun/is_nan.hpp>
 
 static inline void expect_val_deriv_eq(double exp_var, double test_var) {
+  using stan::math::is_nan;
+  if (is_nan(exp_var) && is_nan(test_var)) return;
```

Comment:
Daniel, checking that both the expected and test values are NaN isn't a suggestion. It's what I ultimately implemented in this pull request. I did so by:
1. Using `return` to exit early from functions (the code above) if both expected and test values are NaN
2. Using `continue` in a loop if both expected and test values are NaN
3. If the two options above are not possible, push the check within an `if block`